Mention alternatives to 'none' checkbox option

calvin-lau-sig7 opened the pull request:
## What

Add a sentence to Checkboxes components to tell service teams to consider alternatives to the 'None of these' component, as it may not be the best option for some questions.

1/7/2021: Decision to add an explanation of a 'guard' question and link to it as part of this message.

## Why

Community and working group says that it is important to acknowledge that there are several ways of dealing with the 'none of the above' case and we ought to urge designers to consider different options.

## Who needs to know about this

Content designer, Designer

## Further detail

Discussions in the backlog for [Checkboxes](https://github.com/alphagov/govuk-design-system-backlog/issues/37) component.

## Done when

- [x] Draft sentence in checkboxes component
- [x] Draft explanation of guard question in question page pattern 
- [x] Add example of guard question
- [ ] 2i with team, check with people concerned
- [ ] Publish

CharlotteDowns reviewed (approved):
This looks great, I think we're good to go.

calvin-lau-sig7 commented:
Added new 'done when' to add an explanation of a 'filter' or 'guard' question to the Question page pattern, and link to from Checkbox component. 

Might also need an example.

calvin-lau-sig7 commented:
Drafted an explanation about filter/guard questions, what do we think @CharlotteDowns @joelanman?

Open to other ideas for an example and as discussed, it might be worth adding a code example too.

calvin-lau-sig7 commented:
Hi @CharlotteDowns, would you mind taking another look at this? I've added a change to explain 'filter' questions in the Question page pattern. Afterwards, I'll need to add an anchor link to it.

Joe and I were also considering adding a code example, can you help me make a decision on whether to do?

calvin-lau-sig7 commented on src/patterns/question-pages/index.md.njk:
```diff
@@ -83,6 +83,12 @@ For example, ‘About you’
 
 You can also learn more about how starting with [one thing per page](https://www.gov.uk/service-manual/design/form-structure#start-with-one-thing-per-page) helps users in the GOV.UK Service Manual.
 
+##### Asking 'filter' questions to help users answer a complex question
+
+A series of simple questions can be easier to answer than one complex question. Especially if parts of it aren’t relevant to all users. A 'filter' question will let the user skip parts of a question that do not apply to them so they don't get confused.
+
+For example, you could ask a user if they've travelled outside the UK before asking which countries they've travelled to. This would help a user skip looking at a long list of countries when they do not need to.
```
Okay, slight revision @CharlotteDowns. I didn't realise we reached the bottom of our heading numbers!
```suggestion
#### Asking filter questions simplify a complex question

A series of simple questions can be easier to answer than one complex question. Especially if parts of it aren’t relevant to all users. A 'filter' question will let the user skip parts of a question that do not apply to them so they do not get confused.

For example, you could ask the user if they've travelled outside the UK before you ask them which countries they've travelled to. This would let the user skip looking at a list of countries when they do not need to.
```

CharlotteDowns commented:
Hi Calvin,

I like the revision. I think the written example suffices for now to keep the page length shorter. We can monitor to see if it’s understood correctly.

Charlotte :)

lfdebrux commented on src/patterns/question-pages/index.md.njk:
```diff
@@ -83,6 +83,12 @@ For example, ‘About you’
 
+#### Asking filter questions to simplify a complex question
+
+A series of simple questions can be easier to answer than one complex question. Especially if parts of it aren’t relevant to all users. A 'filter' question will let the user skip parts of a question that do not apply to them so they do not get confused.
```
Bit of a nitpick, but in the other change 'filter question' is in quotes while here it is "'filter' question"; does this inconsistency matter?

calvin-lau-sig7 replied:
Good catch! It's probably should be 'filter question', as it's the term that's misunderstood rather than just the word filter.

calvin-lau-sig7 replied:
```suggestion
A series of simple questions can be easier to answer than one complex question. Especially if parts of it aren’t relevant to all users. A 'filter question' will let the user skip parts of a question that do not apply to them so they do not get confused.
```

lfdebrux reviewed (approved):
One small question, but changes look good to me!

Feel free to merge whenever :)

calvin-lau-sig7 commented:
> One small question, but changes look good to me!
> 
> Feel free to merge whenever :)

Thanks Laurence! Good spot, I've corrected and squashed the extra commit. Merging.
